raspberry/bzzz/sensors/time_of_flight_sensor.py

```diff
@@ -0,0 +1,196 @@
+from bzzz.sensors.filters import median_filter
+from time import time_ns, sleep
+from datetime import datetime
+import VL53L0X
+import os
+
+
+class TimeOfFlightSensor:
+    # NOTE: This class fetches altitude in millimeters and this class requires i2c interfacing with the sensor.
+    def __init__(self,
+                 num_latest_readings_to_keep: int = 3,
+                 update_measurement_at_fixed_rate=False,
```

Comment:
Don't you want it to give you measurements at a constant rate? What is the current update rate? Can it be configured?

Reply:
Currently, we are sampling at `50 Hz` and a `scheduler` is taking care of this constant update rate.  A **custom** update rate cannot be configured by using `update_measurement_at_fixed_rate` because this will only ensure that the sensor is running at the maximum update frequency possible for the current `ranging configuration`. For example, say that the maximum sampling rate is `20 Hz` in `long range mode`, using `update_measurement_at_fixed_rate` only ensures that the sensor is sampled at `20 Hz` (could be less) whilst disabling this and using `scheduler` enables us to sample at **any rate lower than `20 Hz`**.